core/modules/layout/includes/layout.class.inc

```diff
@@ -691,11 +691,24 @@ class Layout {
             $new_positions[$new_region_name][] = $uuid;
           }
           else {
-            unset($this->content[$uuid]);
+            $orphaned_blocks[] = $uuid;
           }
         }
       }
     }
+    if (!empty($orphaned_blocks)) {
+      end($new_positions);
+      $last_region_name = key($new_positions);
+      // Assign orphan blocks to the last region.
```

Comment:
Why the last region? 
We have a default region for each layout, these should be added into the default region instead.

Reply:
Is there a way to know with certainty (via code) what the default region is for a layout? If so, then that makes sense. If not, then we have to pick one randomly, which is why I chose the last region.

Reply:
> Is there a way to know with certainty (via code)

It's defined in the .info file with all the other regions:
```
; Specify regions for this layout.
regions[header]     = Header
regions[top]        = Top
regions[content]    = Content
regions[bottom]     = Bottom
regions[footer]     = Footer

; The default region automatically receives the "Main Content" block if needed.
default region = content
```

For any layout where it's not defined, the fall-back is the region named `content`.

Reply:
Looks like `layout_get_layout_template_info()` will get us the default region.

Reply:
I've updated the PR to use the default region if it exists. If not then it uses the last region. Tests are still not passing so that needs fixing before this is ready.